<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.2">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="1.比对 12345678910111213141516#定义参考基因组路径ref&#x3D;&#x2F;media&#x2F;fqiang&#x2F;TOSHIBA&#x2F;Ubuntu&#x2F;GRCh38&#x2F;GRCh38.bowtie_index#bowtie2进行比对bowtie2 --score-min&#x3D;C,-15,0 --mm -x $ref -q -U SRR957677.fastq.gz 2&gt; bowtie2.log| samtoo">
<meta property="og:type" content="article">
<meta property="og:title" content="circRNA 数据挖掘与分析">
<meta property="og:url" content="http://example.com/2023/04/23/Bioinformatics%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:description" content="1.比对 12345678910111213141516#定义参考基因组路径ref&#x3D;&#x2F;media&#x2F;fqiang&#x2F;TOSHIBA&#x2F;Ubuntu&#x2F;GRCh38&#x2F;GRCh38.bowtie_index#bowtie2进行比对bowtie2 --score-min&#x3D;C,-15,0 --mm -x $ref -q -U SRR957677.fastq.gz 2&gt; bowtie2.log| samtoo">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2023-04-23T14:37:26.799Z">
<meta property="article:modified_time" content="2023-04-23T14:38:02.713Z">
<meta property="article:author" content="John Doe">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="http://example.com/2023/04/23/Bioinformatics%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>circRNA 数据挖掘与分析 | Hexo</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Hexo</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/04/23/Bioinformatics%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="John Doe">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Hexo">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          circRNA 数据挖掘与分析
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2023-04-23 22:37:26 / Modified: 22:38:02" itemprop="dateCreated datePublished" datetime="2023-04-23T22:37:26+08:00">2023-04-23</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>1.比对</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">定义参考基因组路径</span></span><br><span class="line">ref=/media/fqiang/TOSHIBA/Ubuntu/GRCh38/GRCh38.bowtie_index</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">bowtie2进行比对</span></span><br><span class="line">bowtie2 --score-min=C,-15,0 --mm -x $ref -q -U SRR957677.fastq.gz 2&gt; bowtie2.log| samtools view -hbuS - | samtools sort - output.bam</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">分成两部</span></span><br><span class="line">bowtie2 \</span><br><span class="line">--very-sensitive \</span><br><span class="line">--score-min=C,-15,0 \</span><br><span class="line">--mm \</span><br><span class="line">-x $ref -q  \</span><br><span class="line">SRR957677.fastq.gz \</span><br><span class="line"><span class="meta prompt_">&gt;</span><span class="language-bash">SRR.sam\</span></span><br><span class="line"><span class="language-bash">2&gt; bowtie2.log  \</span></span><br><span class="line"><span class="language-bash"></span></span><br><span class="line">cat SRR.sam| samtools view -hbuS |samtools sort -o accepted_hits.bam</span><br></pre></td></tr></table></figure>



<p>2.得到unmapped</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">下载find_circ，里面有很多python程序</span></span><br><span class="line"></span><br><span class="line">samtools view -hf 4 accepted_hits.bam | samtools view -Sb - |./unmapped2anchors.py unmapped.bam | gzip &gt; anchors.fastq.gz</span><br></pre></td></tr></table></figure>



<p>3.找到circ</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 --score-min=C,-15,0 --reorder --mm -q -U anchors.fastq.gz -x $ref |./find_circ.py -G $hg19 &gt; splice_site.bed</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash"><span class="variable">$GRCh38</span> 是参考基因组的序列信息hg19.fa(GRCh38.fa)</span></span><br></pre></td></tr></table></figure>



<p>4.批量下载参考基因组各个染色体（1-22）X，Y，MT（线粒体）</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">for i in $(seq 1 22) X Y MT;</span><br><span class="line">do</span><br><span class="line">ascp -T -l 100M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp.ncbi.nlm.nih.gov --user=anonftp --mode=recv /genomes/H_sapiens/Assembled_chromosomes/seq/hs_ref_GRCh38.p12_chr$&#123;i&#125;.fa.gz ./</span><br><span class="line">done;</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">注意地址以及索引，可能会更新</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">解压所有文件，默认删除压缩文件</span></span><br><span class="line">gunzip *.gz</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">合并所有.fa文件到GRCh38.fa</span></span><br><span class="line">cat *.fa &gt;GRCh38.fa</span><br></pre></td></tr></table></figure>



<ol start="5">
<li><p>fliter the output</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">grep CIRCULAR &lt;run_folder&gt;/splice_sites.bed | \</span><br><span class="line">        grep -v chrM | \#去除线粒体        </span><br><span class="line">        awk &#x27;$5&gt;=2&#x27; | \#at least 2 reads </span><br><span class="line">        grep UNAMBIGUOUS_BP | grep ANCHOR_UNIQUE | \</span><br><span class="line">        ./maxlength.py 100000 \</span><br><span class="line">        &gt; &lt;run_folder&gt;/circ_candidates.bed</span><br></pre></td></tr></table></figure></li>
</ol>
<h2 id=""><a href="#" class="headerlink" title=""></a></h2><h2 id="miRNA测序数据分析"><a href="#miRNA测序数据分析" class="headerlink" title="miRNA测序数据分析"></a>miRNA测序数据分析</h2><h3 id="采用传统比对方法进行分析比对miRNA-seq测序数据"><a href="#采用传统比对方法进行分析比对miRNA-seq测序数据" class="headerlink" title="采用传统比对方法进行分析比对miRNA-seq测序数据"></a>采用传统比对方法进行分析比对miRNA-seq测序数据</h3><p>由miRNA database下载参考数据：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">wget &lt;ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz&gt;  </span><br><span class="line"></span><br><span class="line">wget &lt;ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip&gt;  </span><br><span class="line"></span><br><span class="line">wget &lt;ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip&gt;</span><br><span class="line"></span><br><span class="line">wget &lt;ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3&gt; </span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>将U转换成T</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">perl -alne &#x27;&#123;if(/^&gt;/)&#123;if(/Homo/)&#123;$tmp=1&#125;else&#123;$tmp=0&#125;&#125;;next if $tmp!=1;s/U/T/g if !/&gt;/;print &#125;&#x27; hairpin.fa &gt;hairpin.human.fa</span><br><span class="line">perl -alne &#x27;&#123;if(/^&gt;/)&#123;if(/Homo/)&#123;$tmp=1&#125;else&#123;$tmp=0&#125;&#125;;next if $tmp!=1;s/U/T/g if !/&gt;/;print &#125;&#x27; mature.fa &gt;mature.human.fa</span><br></pre></td></tr></table></figure>

<p>下载原始的测序数据</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir miRNA_test &amp;&amp; cd miRNA_test</span><br><span class="line">for i in `seq 14 19`; do ascp -T -l 100M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp.ncbi.nlm.nih.gov --user=anonftp --mode=recv /sra/sra-instant/reads/ByRun/sra/SRR/SRR154/SRR15427$id/SRR15427$&#123;i&#125;.sra ./; done</span><br></pre></td></tr></table></figure>

<p>批量解压原始测序数据</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *sra|while read id;do fastq-dump $id;done</span><br></pre></td></tr></table></figure>

<p>对测序数据进行质量的过滤和接头的去处</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls *fastq|while read id;do fastq_quality_filter -v -q 20 -p 80 -Q33  -i $id -o tmp;fastx_trimmer -v -f 1 -l 27 -i tmp  -Q33 -z -o $&#123;id%%.*&#125;_clean.fq.gz;</span><br><span class="line">done</span><br><span class="line">rm tmp</span><br></pre></td></tr></table></figure>

<p>建立miRNA的索引包括hairpin，和mature</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">mkdir hairpin &amp;&amp; cd hairpin</span><br><span class="line">bowtie2-build ../hairpin.human.fa hairpin_human</span><br><span class="line">bowtie2-build ../mature.human.fa mature_human</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>进行比对</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls *_clean.fq.gz | while read id ; do  bowtie2 -x ~/miRNA/hairpin/hairpin_human -U $id -S $&#123;id%%.*&#125;.hairpin.sam ; done</span><br><span class="line"></span><br><span class="line">ls *_clean.fq.gz | while read id ; do  bowtie2 -x ~/miRNA/mature/mature_human -U $id -S $&#123;id%%.*&#125;.mature.sam ; done</span><br></pre></td></tr></table></figure>

<p>对比对的sam文件进行计数获得表达量</p>
<h3 id="miRDeep2的使用"><a href="#miRDeep2的使用" class="headerlink" title="miRDeep2的使用"></a>miRDeep2的使用</h3><p>建立物种的基因组索引文件，必须是bowtie建立的索引</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie-build cel_cluster.fa cel_cluster</span><br></pre></td></tr></table></figure>

<p>将测序结果比对map到基因组，-c 可输入fasta格式，-e可输入fastq格式</p>
<p>这里会产生两个重要文件，-s reads_collapsed.fa处理后的reads，-t reads_collapsed_vs_genome.arf这个是mapping之后的结果文件，-j 删除不规范的字母，-k去除接头，-l可接受的最小长度，-p基因组索引，-v将结果显示在屏幕</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl mapper.pl reads.fa -c -j -k TGGAATTCTCGGGTGCCAAGGAACTCCAGTCAC  -l 18 -m -p cel_cluster -s reads_collapsed.fa -t reads_collapsed_vs_genome.arf -v</span><br></pre></td></tr></table></figure>

<p>最重要一步，miRDeep2.pl发现新的miRNA以及对miRNA的表达进行定量，输入的文件是上文处理的文件：reads_collapsed.fa，reads_collapsed_vs_genome.arf，研究物种的成熟miRNA：mature_ref_this_species.fa，其他物种的成熟miRNA：mature_ref_others_species.fa，研究物种的miRNA前体：precursors_ref_this_species.fa。</p>
<p>如果只有reads，arf文件，genome文件，其他文件没有，需要这样表示miRNAs_ref/none miRNAs_other/none  precursors/none，本物种的成熟miRNA无，其他相关物种也无，更没有前体。</p>
<p>-t 研究物种，report.log现实</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">miRDeep2.pl reads_collapsed.fa cel_cluster.fa reads_collapsed_vs_genome.arf</span><br><span class="line"></span><br><span class="line">mature_ref_this_species.fa mature_ref_other_species.fa precursors_ref_this_species.fa -t C.elegans 2&gt;report.log</span><br></pre></td></tr></table></figure>



<h2 id="RNA-seq-测序数据分析"><a href="#RNA-seq-测序数据分析" class="headerlink" title="RNA-seq 测序数据分析"></a>RNA-seq 测序数据分析</h2><p>1.测序数据下载（perfetch)</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 48 62`;do prefetch SRR35899$&#123;i&#125;;done</span><br></pre></td></tr></table></figure>

<p>2.解压数据</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">fastq-dump将.Sra数据解压成.fastq.gz数据，<span class="comment">#--split-3一般是针对双端测序的解压</span></span></span><br><span class="line">for i in `seq 56 62`;</span><br><span class="line">do</span><br><span class="line"> fastq-dump --gzip --split-3 -O ~/Data/RNA-seq -A SRR35899$&#123;i&#125;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>3,fastqc 测序数据质量报告</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc SRR35899$&#123;i&#125;.fastq.gz</span><br></pre></td></tr></table></figure>

<p>4,下载index（进行序列比对的索引）</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">cd reference &amp;&amp; mkdir index &amp;&amp; cd index</span><br><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz</span><br><span class="line">tar -zxvf hg19.tar.gz</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">如果是bowtie2比对就要下载.bt结尾的索引</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">如果是hisat2比对就要下载.ht结尾的索引</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">这里建议直接去官网下载建立好的索引，一般电脑更能不能顺利建立索引</span></span><br></pre></td></tr></table></figure>

<p>5,对测序结果进行比对</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p RNA-Seq/aligned</span><br><span class="line">for i in `seq 56 58`</span><br><span class="line">do</span><br><span class="line"> hisat2 -t -x reference/index/hg19/genome -1 RNA-Seq/SRR35899$&#123;i&#125;_1.fastq.gz -2 SRR35899$&#123;i&#125;_2.fastq.gz -S RNA-Seq/aligned/SRR35899$&#123;i&#125;.sam </span><br><span class="line">done</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">-x 后是参考基因组所以的路径，一般是把各个索引相同的前缀部分写完</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">hisat2的hg19索引是genome.1.ht2,genomme.2.ht2，所以索引路径就要写词path/hg19/genome</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">如果不是双端测序数据，-1，-2改成-U就可以了</span></span><br></pre></td></tr></table></figure>

<p>6,SAM tools进行转换，排序，建立index(可选）</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 56 58`</span><br><span class="line">do</span><br><span class="line"> samtools view SRR35899$&#123;i&#125;.sam -b &gt; SRR35899$&#123;i&#125;.bam</span><br><span class="line"> samtools sort SRR35899$&#123;i&#125;.bam -o SRR35899$&#123;i&#125;_sorted.bam</span><br><span class="line"> samtools index SRR35899$&#123;i&#125;_sorted.bam</span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>#7，可采用RSeQC对比对结果进行质控（可选步骤，依赖于python）<br>bam_stat.py -i SRR3589956_sorted.bam</p>
<p>7，reads计数</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 56 58`</span><br><span class="line">do</span><br><span class="line">htseq-count -s no -r pos -f bam RNA-Seq/aligned/SRR35899$&#123;i&#125;_sorted.bam reference/gencode.v26lift37.annotation.sorted.gtf &gt; RNA-Seq/matrix/SRR35899$&#123;i&#125;.count 2&gt; RNA-Seq/matrix/SRR35899$&#123;i&#125;.log</span><br><span class="line">done</span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">-s no/yes/reverse数据是否来自strand-specific assay，如果是no，每一条<span class="built_in">read</span>都会和两条链比较</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">-r name/pos 利用samtool <span class="built_in">sort</span>对数据是根据name排序还是pos排序，默认name</span></span><br><span class="line"><span class="meta prompt_">#</span><span class="language-bash">-f bam/sam 输入文件的格式</span></span><br></pre></td></tr></table></figure>

<p>8，R语言对counts进行差异分析DESeq2</p>

    </div>

    
    
    

      <footer class="post-footer">

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2023/04/23/hello-world/" rel="prev" title="Hello World">
      <i class="fa fa-chevron-left"></i> Hello World
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link"><span class="nav-number">1.</span> <span class="nav-text"></span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#miRNA%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90"><span class="nav-number">2.</span> <span class="nav-text">miRNA测序数据分析</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E9%87%87%E7%94%A8%E4%BC%A0%E7%BB%9F%E6%AF%94%E5%AF%B9%E6%96%B9%E6%B3%95%E8%BF%9B%E8%A1%8C%E5%88%86%E6%9E%90%E6%AF%94%E5%AF%B9miRNA-seq%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE"><span class="nav-number">2.1.</span> <span class="nav-text">采用传统比对方法进行分析比对miRNA-seq测序数据</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#miRDeep2%E7%9A%84%E4%BD%BF%E7%94%A8"><span class="nav-number">2.2.</span> <span class="nav-text">miRDeep2的使用</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#RNA-seq-%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90"><span class="nav-number">3.</span> <span class="nav-text">RNA-seq 测序数据分析</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">John Doe</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">2</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">John Doe</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
